src/clients/Deluge.ts

```diff
@@ -263,43 +285,48 @@ export default class Deluge implements TorrentClient {
 				return InjectionResult.FAILURE;
 			}
 
+			const torrentFileName = `${newTorrent.getFileSystemSafeName()}.cross-seed.torrent`;
+			const encodedTorrentData = newTorrent.encode().toString("base64");
+			const torrentPath = path ? path : torrentInfo!.save_path;
 			const params = this.formatData(
-				`${newTorrent.getFileSystemSafeName()}.cross-seed.torrent`,
-				newTorrent.encode().toString("base64"),
-				path ? path : torrentInfo!.save_path,
-				!!searchee.infoHash,
+				torrentFileName,
+				encodedTorrentData,
+				torrentPath,
 				decision,
 			);
+
 			const addResult = await this.call<string>(
 				"core.add_torrent_file",
 				params,
 			);
-			if (addResult.result) {
-				const { linkingCategory } = getRuntimeConfig();
+
+			const addResponse =
+				typeof addResult.error?.message === "string"
+					? addResult.error
+					: addResult.result;
+
+			if (typeof addResponse === "string") {
```

Comment:
I think the part that worries me about this specifically is that `typeof addResponse === "string"` really semantically means "the operation succeeded", but when you read `typeof addResponse === "string"`, nothing in the expression itself indicates any notion of success/failure.

Reply:
the result-deluge branch solves this, so it's kind of irrelevant for this PR to readdress it that way when there's already working code using Result...

Reply:
where is that branch? can it be merged into this one?

Reply:
it could be... its currently the last in the chain of like 6 lol

Reply:
ah never mind then.

Reply:
So they only place where Deluge.ts is modified after this branch in the "chain" before the result branch is in the "Ensemble matching" commit/branch. 

It could easily target this with some changes, but it would need some minor changes - specifically getDownloadDir() and getAllDownloadDirs() and a labeling return that was incompatible with virtual searchee's.

It is not major, and I could see the value in getting that code in sooner rather than later.

If you want me to do this, I can figure it out rather easily. We just need to make sure that it's handled properly in the ensemble branch which is downstream from this one (this branch -> arr-ids -> ensemble -> inject-saved-torrents -> filter-search-logging -> result-deluge)

Being that it's entirely focused in one commit, it shouldn't be a big deal.

Let me know.